### ISSUE TITLE
Add a field to an existing section

### ISSUE THREAD

**Issue by nicolas92** [reduxframework/redux-framework#2992]
Hello, I have a little problem, I would like to add a field to an existing section.
Here is my code section in the parent theme:

<pre>Redux::setSection( $opt_name, array(
    'id'            => 'widgets',
    'icon'          => 'el el-cogs',
    'title'         => __( 'Widgets', 'kho' ),
    'customizer'    => false,
    'fields'        => array(
        array(
            'id'        => 'about-me',
            'type'      => 'switch', 
            'title'     => __( 'About Me', 'kho' ),
            "default"   => '1',
            'on'        => __( 'On', 'kho' ),
            'off'       => __( 'Off', 'kho' ),
        ),
    ),
));</pre>

I would like to add a new field in this section (widgets) via my child theme, is it possible?
I tried the following code but it does not work:

<pre>add_filter( 'redux/options/uwl_options/sections', 'new_widgets_fields' );
function new_widgets_fields( $sections ) {

    $sections['widgets'] = array(
       'fields'        => array(
            array(
                'id'        => 'test',
                'type'      => 'switch', 
                'title'     => __( 'Test', 'kho' ),
                "default"   => '1',
                'on'        => __( 'On', 'kho' ),
                'off'       => __( 'Off', 'kho' ),
            )
        )
    );

    return $sections;

}</pre>

I looked in the documentation, on internet and in the issue on github, nothing works, I managed to create a new section, but I would add a new field to an existing section.

Can you help me?

Thank you a lot :)

**Comment by kprovance** [comment]
You can't use the Redux API and hook that utlize the legacy method to add fields.  They won't work together.

See:  https://docs.reduxframework.com/core/redux-api/

You want setField.

**Comment by dovy** [comment]
Well, you can but it's untested. Hah. Use the API. ;)

**Comment by nicolas92** [comment]
Hello, thank you for your answers, I tried the following code but it still does not work:

<pre>add_filter( 'redux/options/uwl_options/register', 'uwl_pro_widgets_fields' );
function uwl_pro_widgets_fields( $fields ) {

    $opt_name = 'uwl_options';

    $field = array(
        'id'       => 'opt-text-example',
        'type'     => 'text',
        'title'    => 'Text Field',
        'subtitle' => 'Subtitle',
        'desc'     => 'Field Description',
        'default'  => 'Default Text',
        'section_id' => 'widgets',
    );

    Redux::setField( $opt_name, $field );

}</pre>

This may be due to the filter?

Thank you.

**Comment by kprovance** [comment]
We've already told you what you need to do.

**Comment by nicolas92** [comment]
Sorry but I have not understood, I looked at the Redux API, I modified my code but it still does not work.
Can you show me a code example?
I looked in the sample file, there is no example to add a new field to an existing section.
thank you.

**Comment by kprovance** [comment]
I'd have to have set up on my side an exact replica of your situation, and that would take away from time I need to work at my paying job.  if you'd like to purchase some premium support, then I could spend an hour on it.  I know it works.  I've not done it from a child theme, but I have from a plugin, added sections to a theme.  Your fields look correct, but I think it's the hook not working.  It may be running too late

What I did, in my theme, was load my options on the after_theme_setup hook, with a priority of 3.  IN my plugin, where I added a section to the panel I loaded on the after_theme_setup hook with a priority of 1.  ALL of this is contingent upon Redux being already loaded.  If it isn't, none of this will work.
